test/docfx.Test/build/TocTest.cs

```diff
@@ -123,18 +123,30 @@ public static void TocParserLoadMarkdownToc()
         [Fact]
         public static void TocParserLoadBadMarkdownToc()
         {
-            var ex = Assert.Throws<FormatException>(() =>
-                TableOfContentsParser.LoadMdTocModel(@"
-#[good](test.md)
+            var (errors, model) = MarkdownTocMarkup.LoadMdTocModel(@"
+# [good](test.md)
 [bad]()
 >_<
 >_<
 >_<
-", "TOC.md"));
-            Assert.Equal(@"Invalid toc file, FilePath: TOC.md, Details: Unknown syntax at line 3:
-[bad]()
->_<
->_<".Replace("\r\n", "\n"), ex.Message.Replace("\r\n", "\n"));
+", "TOC.md");
+
+            Assert.Equal(2, errors.Count);
```

Comment:
>Assert.Equal(2, errors.Count); [](start = 12, length = 30)

We can use Assert.Collection(errors,) to verify error count and content at the same time